### ISSUE TITLE
Cannot retrive loading status

### ISSUE THREAD

**Issue by Mowinski** [ilteoood/flutter_i18n#222]
Hi, I want to react to loading status on my code and display loading popup when user change language.

In init state I have:

```
  @override
  void initState() {
    super.initState();
     Future.delayed(Duration.zero, () {
      FlutterI18n.retrieveLoadedStream(context).listen((event) {
        log(LogType.main, LogLevel.INFO, {"message": "OnChange", "event": event.toString()});
      });
    });
  }
  ```

I use my custom TranslationDelegate, and custom loading method

```
LocalizationsDelegate<dynamic> getRewardLocalizationsDelegate() {
  return FlutterI18nDelegate(
    translationLoader: CyberbiegRewardTranslationLoader(),
  );
}

class CyberbiegRewardTranslationLoader extends TranslationLoader {
  @override
  Future<Map> load() async {
    String languageCode = "pl";
    if (locale != null) {
      languageCode = locale!.languageCode;
    }
    final translationService = getIt<TranslationService>();
    final result = translationService
        .getTranslation()
        .then((translation) => saveLanguageFile(translation, languageCode))
        .catchError(onError);

    log(LogType.main, LogLevel.INFO, {
      "message": "Loaded reward translations from server",
      "class": "TranslationLoader",
    });
    return result;
  }
}
```

And load in `CyberbiegRewardTranslationLoader` is invoked but in listen I have no events. But language change correctly and all logs from load method are displayed. In router I have standard code:

```
localizationsDelegates: [
          getRewardLocalizationsDelegate(),
          ...AppLocalizations.localizationsDelegates,
        ],
```

**Comment by ilteoood** [comment]
Hi,
the stream that you are listening to is based on a simple `StreamController`.
It means that all the events already fired are not re-sent. So probably everything is already loaded when you try to listen for that event.
I'm thinking to replace it with a `BehaviorSubject`.

**Comment by ilteoood** [comment]
I think this has been fixed with version 0.36.1
